### ISSUE TITLE
Behaviour of Masking layer with LSTM

### ISSUE THREAD

**Issue by stillbreeze** [fchollet/keras#7227]
Please make sure that the boxes below are checked before you submit your issue. If your issue is an implementation question, please ask your question on [StackOverflow](http://stackoverflow.com/questions/tagged/keras) or [join the Keras Slack channel](https://keras-slack-autojoin.herokuapp.com/) and ask there instead of filing a GitHub issue.

Thank you!

- [x] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps

- [x] If running on TensorFlow, check that you are up-to-date with the latest version. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [x] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

- [x] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

Why is the behaviour of the Dense layer different with masking than the LSTM layer?
Ideally, the timestep (axis 1) which was masked should output 0, but that is not the case when using Masking with LSTMs.

Any idea what's going on?

A small script for comparing the behaviour of LSTM and Dense units with Masking:

```
import numpy as np
from keras.models import Model
from keras.layers import Input, LSTM, Masking, TimeDistributed, Dense

inp_phrase_1 = Input(shape=(15, 10))
mask_1 = Masking(mask_value=0.0)(inp_phrase_1)
lstm = LSTM(10, return_sequences=True)(mask_1)
model_1 = Model(inputs=[inp_phrase_1], outputs=[lstm])

inp_phrase_2 = Input(shape=(15, 10))
mask_2 = Masking(mask_value=0.0)(inp_phrase_2)
dense = TimeDistributed(Dense(10))(mask_2)
model_2 = Model(inputs=[inp_phrase_2], outputs=[dense])

X = np.random.rand(1,15,10)
X[0,10,:] = 0.0

o1 = model_1.predict(X)
o2 = model_2.predict(X)
print o1[0,10,:]
print o2[0,10,:]
```
Output:
```
[ 0.17980759 -0.19799934  0.08030579  0.09331396  0.43052146 -0.20179336
 -0.33242977  0.28832835 -0.12895377  0.35023171]
[ 0.  0.  0.  0.  0.  0.  0.  0.  0.  0.]
```
@farizrahman4u : Would be really glad if you could help out with this. Is this behaviour expected?

**Comment by prasadkawthekar** [comment]
Hi, did you figure out what was happening?
